### PR TITLE
Updated testnet for ticket "New official Hush testnet #bounty #6"

### PR DESCRIPTION
**Pull request by mrhcbojangles**
Updated the testnet parameters to "tush"
Generated/mined a new genesis block
Added the host info into the block for the origin testnet code currently running at tush.cryptominingpools.org.

Please review.

Thanks,

Tim

tAddress: t1c11VazKq24xJoM55Cp4rpxeM4p6xxmcuR
**edit madbuda**
fixes https://github.com/MyHush/hush/issues/6

### CODE REVIEW

**Comment by madbuda** [comment]
fixes https://github.com/MyHush/hush/issues/6

**Comment by madbuda** [comment]
test this please

**Comment by leto** [comment]
:+1: from me

**Comment by leto** [comment]
@radix42 does this meet your requirements? If so, let's pay out the bounty and get our testnet up so we can play with it!

**Comment by radix42** [comment]
can we get a couple/few testnet nodes on it? if so and they can
send/receive txns I'll very happily pay out and close this!

**Comment by madbuda** [comment]
We also need to test / cherry pick this into 1.0.12 and verify testnet with that release

**Comment by leto** [comment]
@madbuda should we merge this into a topic branch off `dev` so we can more easily run a few nodes with this code, to test it out?

**Comment by madbuda** [comment]
There is no need, just build from https://github.com/mrhcbojangles/hush

We do need some edits for long term usage, primarily seed nodes

@mrhcbojangles please add testnet.myhush.network in chainparams

File: src/chainparams.cpp [hunk 1]
```diff
@@ -223,16 +223,16 @@ class CTestNetParams : public CMainParams {
...
         consensus.hashGenesisBlock = genesis.GetHash();
         assert(consensus.hashGenesisBlock == uint256S("0x" + consensus.hashGenesisBlock.ToString()));
 
         vFixedSeeds.clear();
         vSeeds.clear();
-        vSeeds.push_back(CDNSSeedData("z.cash", "dnsseed.testnet.z.cash")); // Zcash
+        vSeeds.push_back(CDNSSeedData("tush1", "tush.cryptominingpools.org")); // Zcash
```
Comment (madbuda) [inline src/chainparams.cpp]: need to add
`vSeeds.push_back(CDNSSeedData("node1", "testnet.myhush.network")); // madbuda`

Comment (mrhcbojangles) [reply src/chainparams.cpp]: Added testnet.myhush.network to chainparams. I kept the tush.cryptominingpools.org there until there are more nodes. I can remove it if you would like me to.

Comment (mrhcbojangles) [reply src/chainparams.cpp]: I will rebuild the node on tush.cryptominingpools.org this evening so it will align with the latest on the forked branch.

**Comment by madbuda** [comment]
not sure if the node listed in this PR is stil live, I am getting crashes generating new blocks
hushd: wallet/wallet.cpp:803: void CWallet::DecrementNoteWitnesses(const CBlockIndex*): Assertion `nWitnessCacheSize > 0' failed.
Aborted

File: src/chainparams.cpp (hunk 1 above)
Comment (madbuda) [reply src/chainparams.cpp]: Thanks, have you tried using the CPU miner? I don't see any blocks and inning crashes hushd. I would assume that the first nod will need to get difficulty up a little before others can be stable

**Comment by mrhcbojangles** [comment]
I updated the config on the node to include testnet.myhush.network as well. The block height is currently at 85, so blocks are being generated. I would try to add -reindex and try again.

**Comment by madbuda** [comment]
Ok, I'll try again when I get home. Is the port open and node running with server=1 ?

**Comment by madbuda** [comment]
test this please

File: src/chainparams.cpp (hunk 1 above)
Comment (mrhcbojangles) [reply src/chainparams.cpp]: The CPU miner is enabled, it requires connections before it starts. When testing, I connected another node to it and mined 10 or 15 blocks. The block height is currently at 85. I do see where it failed to connect to 192.119.14.214 (testnet.myhush.network), but I believe that is just where it wasn't online. I opened up all of the ports to that VM, so try to run it again with -reindex and see if you still get the error.

**Comment by mrhcbojangles** [comment]
My config looks like:
rpcuser=<username>
rpcpassword=<password>
rpcport=<port>
testnet=1
gen=1
genproclimit=1
server=1
equihashsolver=tromp
addnode=testnet.myhush.network

**Comment by mrhcbojangles** [comment]
I should clarify that I left them blank here, but the username password and port are not blank in the config. Sorry for any confusion.

**Comment by madbuda** [comment]
what is the p2p port? And is that port open to the world?

**Comment by mrhcbojangles** [comment]
The p2p port is 18888 (unchanged). It was open to the world. I will reconnect another node to it when I get home to see if I can reproduce what you are seeing.

**Comment by madbuda** [comment]
@mrhcbojangles I reindexed and let it sync before mining and it looks good so far. I will spin up a few more nodes

**Comment by mrhcbojangles** [comment]
Awesome. I see that the block count has gone up as well. Let me know if you run into anything else.

**Comment by madbuda** [comment]
Lets get some confirmations and some tx sent and I will give my 👍🏼
Still need to get a few more nodes up and possibly a testnet explorer if I have time

**Comment by madbuda** [comment]
https://github.com/madbuda/hush/tree/v1.0.9-win
merged in dev and testnet to get a windows client up.

Haven't been able to get a transaction to confirm using linux, dropped new exe's into hush swing wallet to see behavior better

This seems odd to me, blocks are created by a new address every time, will have to wait for blocks to confirm which may take some time
hush-testnet

File: src/chainparams.cpp (hunk 1 above)
Comment (madbuda) [reply src/chainparams.cpp]: vSeeds.push_back(CDNSSeedData("node2", "testnet.madmining.club")); // madmining

another permanent testnet node

**Comment by mrhcbojangles** [comment]
I updated my config to include mineraddress=<taddr>. Is it possible that the node generates an address if one is not specified? This would explain the new addresses. Also, how many confirmations do the transactions have? They will show up immature for a few hours after to obtain the 100 confirmations.

**Comment by madbuda** [comment]
100 confirmations are needed for a block, you must shield (send from miner address to a z_address) before the coinbase funds can be spent (T to T)

See my edit above, I missed that originally

**Comment by madbuda** [comment]
test this please

File: src/chainparams.cpp (hunk 1 above)
Comment (mrhcbojangles) [reply src/chainparams.cpp]: I've updated the fork with the madmining url.

**Comment by mrhcbojangles** [comment]
@madbuda Any luck getting the transactions to confirm?

**Comment by madbuda** [comment]
yes, I have several confirmed tx, TtoZ, TtoT, ZtoZ, ZtoT.

**Comment by madbuda** [comment]
@radix42 @leto approved, I have done testing and approved as well.

**Comment by radix42** [comment]
Awesome, I can pay out bounty in a while this morning

**Comment by madbuda** [comment]
ok, I will merge and get a build up and a PR for master

**Comment by radix42** [comment]
Bueno, thank you

**Pull request merged**